### DISCUSSION TITLE
Setting length of the IDs

### DISCUSSION THREAD

**Discussion by whobutsb** [spatie/laravel-prefixed-ids#13] [Ideas]
Would you be open to being able to set the length of the IDs for the prefixed ID in the `config/prefixed_ids.php`?  

By default in the config.php it would be set to:
```
<?php

return [
    /*
     * The attribute name used to store prefixed ids on a model
     */
    'prefixed_id_attribute_name' => 'prefixed_id',

    /*
     * Set the length of the prefixed ids for your model.
     * Setting null will use laravel's Str::uuid helper
     */
    'prefixed_id_length' => null,
];
```

If an integer is set for `prefixed_id_length` the `getUniquePartForPrefixId()` method would be updated to:
```
    protected function getUniquePartForPrefixId(): string
    {
        if($length = config('prefixed-ids.prefixed_id_length')){
            return substr(md5(uniqid(mt_rand(), true)), 0, $length);
        }

        return str_replace('-', '', Str::uuid());
    }
```

We could even setup an exception to check to make sure the `prefixed_id_length` > 8 so the risk of collision is less likely.  

Just opening the conversation because for some of the models with prefixed_ids I would like to use them as short urls, `https://example.com/l/camp_3b3ka5k48`.  

Thank you for the consideration.

**Comment by freekmurze** [comment]
Thanks for your proposal. I'd rather have a more flexible solution.

We could give users the possibility of passing a closure that returns the unique part of the id. In usage it could look like this.

```
PrefixedIds::generateUniqueIdUsing(function(Model $model) {
   return rand(); // returns something random
});
```

Users can this way return unique parts of any length they desire.

I'd accept a PR that adds this. Make sure to update the readme and include tests.
